File: 2713-find-the-divisibility-array-of-a-string/find-the-divisibility-array-of-a-string.cpp
class Solution {
public:
    vector<int> divisibilityArray(string word, int m) {
        const unsigned int wl = word.length();
        vector<int> ans;
        long num = 0;
        for (unsigned int i = 0; i < wl; i++) {
            num *= 10;
            num += word[i] - '0';
            num %= m;
            ans.push_back(num == 0);
        }
        return ans;
    }
};

File: 2713-find-the-divisibility-array-of-a-string/find-the-divisibility-array-of-a-string.py
class Solution:
    def divisibilityArray(self, word: str, m: int) -> List[int]:
        ans = [0] * len(word)
        curr = 0
        for i, c in enumerate(word):
            curr = curr*10 + int(c)
            if curr % m == 0:
                ans[i] = 1
            curr = curr % m
        return ans